Bluetooth: host: Expose per adv evt counter in CTE IQ report.

jakkra opened the pull request:
The periodic advertising event counter is present in the `HCI_LE_Connectionless_IQ_Report` but until now not availible for the application to access. This pull request fixes that.

Signed-off-by: Jakob Krantz <jakob.krantz@u-blox.com>

jakkra committed:
Bluetooth: host: Expose per adv evt counter in CTE IQ report.

Allows the application to access the periodic advertising counter.

Signed-off-by: Jakob Krantz <jakob.krantz@u-blox.com>

ppryga-nordic reviewed (approved):
LGTM

Thalley commented on subsys/bluetooth/host/direction.c:
```diff
@@ -351,6 +351,7 @@ void hci_df_prepare_connectionless_iq_report(struct net_buf *buf,
 	report->cte_type = BIT(evt->cte_type);
 	report->packet_status = evt->packet_status;
 	report->slot_durations = evt->slot_durations;
+	report->per_evt_counter = evt->per_evt_counter;
```
```suggestion
	report->per_evt_counter = sys_le16_to_cpu(evt->per_evt_counter);
```

This may also be missing for other fields

jakkra replied:
Thanks, added this.

ppryga replied:
Good catch @Thalley thank you!

Thalley replied:
@ppryga @jakkra Looks like the RSSI is missing a `sys_le16_to_cpu`, will one of you add a PR to fix that one, or add it to this PR?

jakkra replied:
Handled for RSSI in a new commit.

jhedberg commented on include/bluetooth/direction.h:
```diff
@@ -101,6 +101,8 @@ struct bt_df_per_adv_sync_iq_samples_report {
 	/** Status of received PDU with CTE (@ref bt_df_packet_status). */
 	uint8_t packet_status;
+	/** Value of the paEventCounter of the AUX_SYNC_IND PDU. */
+	uint16_t per_evt_counter;
```
Unrelated to this PR, but someone should look into the ordering of struct members here. It seems like there might be unnecessary padding because of it.

ppryga replied:
I'll take a look on this. Thanks @jhedberg

jakkra committed:
Bluetooth: host: Add endian handling of rssi in adv. IQ report.

When populating bt_df_per_adv_sync_iq_samples_report the
rssi was not wrapped in sys_le16_to_cpu.

Signed-off-by: Jakob Krantz <jakob.krantz@u-blox.com>
